backend/app/main.py

```
@@ -241,11 +241,30 @@ async def _startup_watch():
-        search_index.rebuild()
-    except Exception:
-        pass
-    
+
+    # Rebuild search index in background thread with timeout to prevent startup hang
+    def _rebuild_with_timeout():
+        try:
+            import signal
+            # Use alarm on Unix, skip timeout on Windows
+            if hasattr(signal, 'SIGALRM'):
+                def timeout_handler(signum, frame):
```

Ruff (0.14.13)

Unused function argument: `signum`

(ARG001)